File: src/types/visualInspectionType.ts
import { LocationType } from './locationType';
import { VisualInspectionKeyType } from '../enums/visualInspectionKeyType';
import { ValueType } from './valueType';

export type VisualInspectionType = {
  [VisualInspectionKeyType.Img]: string;
  [VisualInspectionKeyType.DamageType]: string;
  [VisualInspectionKeyType.Date]: string;
  [VisualInspectionKeyType.UserId]: string;
  [VisualInspectionKeyType.Location]: LocationType;
};

export const VisualInspectionValueMap: Record<VisualInspectionKeyType, ValueType> = {
  [VisualInspectionKeyType.Img]: 'string',
  [VisualInspectionKeyType.DamageType]: 'string',
  [VisualInspectionKeyType.Date]: 'string',
  [VisualInspectionKeyType.UserId]: 'string',
  [VisualInspectionKeyType.Location]: 'LocationType'
};

File: backend/slab2reuse/src/main/java/circrete/slab2reuse/model/VisualInspectionType.java
package circrete.slab2reuse.model;

import jakarta.persistence.*;

@Entity
public class VisualInspectionType {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String img;
    private String damageType;
    private String date;
    private String userId;

    @OneToOne(cascade = CascadeType.ALL)
    private LocationType location;

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    public String getDamageType() {
        return damageType;
    }

    public void setDamageType(String damageType) {
        this.damageType = damageType;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public LocationType getLocation() {
        return location;
    }

    public void setLocation(LocationType location) {
        this.location = location;
    }
}
